praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/service/PraktikumsstellenService.java

```diff
@@ -149,6 +158,105 @@ public void deleteAusbildungsPraktikumsstelle(UUID praktikumsstellenId) {
         }
     }
 
+    public void updateAusbildungsPraktikumsstelle(UUID praktikumsstellenId,
+            UpdateAusbildungsPraktikumsstelleWithMeldezeitraumAndAssignedNWKDto praktikumsstelleDto) {
+
+        AusbildungsPraktikumsstelle ausbildungsPraktikumsstelle = praktikumsstellenMapper.toEntity(praktikumsstellenId, praktikumsstelleDto);
+        ausbildungsPraktikumsstelle.setDienststelle(normalizeDienststelle(ausbildungsPraktikumsstelle.getDienststelle()));
+
+        if (praktikumsstelleDto.assignedNwk() != null) {
+            updateAusbildungsPraktikumsstelleWithAssignedNwk(praktikumsstellenId, praktikumsstelleDto);
+        } else if (ausbildungsPraktikumsstellenRepository.existsById(praktikumsstellenId)) {
+            ausbildungsPraktikumsstellenRepository.save(ausbildungsPraktikumsstelle);
+        } else {
+            throw new ResourceNotFoundException("Praktikumsstelle nicht gefunden.");
+        }
+    }
+
+    public void updateStudiumsPraktikumsstelle(UUID praktikumsstellenId, UpdateStudiumsPraktikumsstelleWithMeldezeitraumAndAssignedNwkDto praktikumsstelleDto) {
+
+        StudiumsPraktikumsstelle studiumsPraktikumsstelle = praktikumsstellenMapper.toEntity(praktikumsstellenId, praktikumsstelleDto);
+        studiumsPraktikumsstelle.setDienststelle(normalizeDienststelle(studiumsPraktikumsstelle.getDienststelle()));
+
+        if (praktikumsstelleDto.assignedNwk() != null) {
+            updateStudiumsPraktikumsstelleWithAssignedNwk(praktikumsstellenId, praktikumsstelleDto);
+        } else if (studiumsPraktikumsstellenRepository.existsById(praktikumsstellenId)) {
+            studiumsPraktikumsstellenRepository.save(studiumsPraktikumsstelle);
+        } else {
+            throw new ResourceNotFoundException("Praktikumsstelle nicht gefunden.");
+        }
+    }
+
+    private void updateAusbildungsPraktikumsstelleWithAssignedNwk(UUID id,
+            UpdateAusbildungsPraktikumsstelleWithMeldezeitraumAndAssignedNWKDto praktikumsstelleDto) {
+        Optional<AusbildungsPraktikumsstelle> ausbildungsPraktikumsstelleOptional = ausbildungsPraktikumsstellenRepository.findById(id);
+        if (ausbildungsPraktikumsstelleOptional.isEmpty()) {
+            throw new ResourceNotFoundException("Praktikumsstelle nicht gefunden.");
+        }
```

Comment:
could be extracted to a separate method `findOrThrow` that returns an object of class `AusbildungsPraktikumsstelle`. If not the error message should be a constant that is reused.